### ISSUE TITLE
Issues in installing Genie on Julia 1.7.2 

### ISSUE THREAD

**Issue by gameCoder101** [GenieFramework/Genie.jl#533]
**Describe the bug**
After installing Genie, I have this error:

Precompiling project...
  ✗ Nettle_jll
  ✗ Nettle
  ✗ Genie
  14 dependencies successfully precompiled in 20 seconds (377 already precompiled)
  3 dependencies errored. To see a full report either run `import Pkg; Pkg.precompile()` or load the packages

(@v1.7) pkg> 

**To reproduce**
pkg> add Genie

**Additional context**
Please include the output of 
`julia> Julia Version 1.7.2
Commit bf53498635 (2022-02-06 15:21 UTC)
Platform Info:
  OS: Linux (x86_64-redhat-linux) --> Fedora 36
  CPU: Intel(R) Core(TM) i7-4702HQ CPU @ 2.20GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-12.0.1 (ORCJIT, haswell)()`

**Comment by essenciary** [comment]
@gameCoder101 Looks like precompiling Nettle (a dependency) fails. Can you try adding/using/precompiling Nettle separately to see what crashes it?

**Comment by gameCoder101** [comment]
I tried to install Nettle this morning and everything seems fine.
I uninstalled Netlle and Genie. I installed Nettle first - OK.
Installed Genie, I get some errors:
(@v1.7) pkg> add Nettle
    Updating registry at `~/.julia/registries/General.toml`
   Resolving package versions...
    Updating `~/.julia/environments/v1.7/Project.toml`
  [49dea1ee] + Nettle v0.5.1
    Updating `~/.julia/environments/v1.7/Manifest.toml`
  [49dea1ee] + Nettle v0.5.1
  [4c82536e] + Nettle_jll v3.7.2+0
  [781609d7] + GMP_jll

(@v1.7) pkg> add Genie
   Resolving package versions...
    Updating `~/.julia/environments/v1.7/Project.toml`
  [c43c736e] + Genie v4.18.1
    Updating `~/.julia/environments/v1.7/Manifest.toml`
  [c7e460c6] + ArgParse v1.1.4
  [8f5d6c58] + EzXML v1.1.0
  [c43c736e] + Genie v4.18.1
  [77172c1b] + HttpCommon v0.5.0
  [6d011eab] + Inflector v1.0.1
  [6f1432cf] + LoweredCodeUtils v2.2.2
  [39ec1447] + Millboard v0.2.5
  [4d1e1d77] + Nullables v1.0.0
  [295af30f] + Revise v3.3.3
  [69024149] + StringEncodings v0.3.5
  [b718987f] + TextWrap v1.0.1
  [856ac37a] + UrlDownload v1.0.0
  [a637dc6b] + VersionCheck v0.2.0
  [ddb6d928] + YAML v0.4.7
Precompiling project...
  ✗ Genie
  0 dependencies successfully precompiled in 9 seconds (391 already precompiled, 2 skipped during auto due to previous errors)
  1 dependency errored. To see a full report either run `import Pkg; Pkg.precompile()` or load the package
-------------

julia> import Genie; Genie.precompile()
[ Info: Precompiling Genie [c43c736e-a2d1-11e8-161f-af95117fbd1e]
ERROR: LoadError: InitError: could not load library "libgmpxx.so.4"
libgmpxx.so.4: cannot open shared object file: No such file or directory
Stacktrace:
 [1] __init__()
   @ GMP_jll /usr/share/julia/stdlib/v1.7/GMP_jll/src/GMP_jll.jl:36
 [2] top-level scope
   @ ~/.julia/packages/JLLWrappers/QpMQW/src/toplevel_generators.jl:188
 [3] top-level scope (repeats 2 times)
   @ none:1
during initialization of module GMP_jll
in expression starting at /home/username/.julia/packages/Nettle_jll/LjWzs/src/wrappers/x86_64-linux-gnu.jl:4
in expression starting at /home/username/.julia/packages/Nettle_jll/LjWzs/src/Nettle_jll.jl:2
ERROR: LoadError: Failed to precompile Nettle_jll [4c82536e-c426-54e4-b420-14f461c4ed8b] to /home/username/.julia/compiled/v1.7/Nettle_jll/jl_e7qTyy.
Stacktrace:
 [1] top-level scope (repeats 2 times)
   @ none:1
in expression starting at /home/username/.julia/packages/Nettle/NhQ2s/src/Nettle.jl:1
ERROR: LoadError: Failed to precompile Nettle [49dea1ee-f6fa-5aa6-9a11-8816cee7d4b9] to /home/username/.julia/compiled/v1.7/Nettle/jl_Q995RU.
Stacktrace:
 [1] include(x::String)
   @ Genie ~/.julia/packages/Genie/hbOGK/src/Genie.jl:4
 [2] top-level scope
   @ ~/.julia/packages/Genie/hbOGK/src/Genie.jl:34
 [3] top-level scope (repeats 2 times)
   @ none:1
in expression starting at /home/username/.julia/packages/Genie/hbOGK/src/Encryption.jl:1
in expression starting at /home/username/.julia/packages/Genie/hbOGK/src/Genie.jl:1
ERROR: Failed to precompile Genie [c43c736e-a2d1-11e8-161f-af95117fbd1e] to /home/username/.julia/compiled/v1.7/Genie/jl_rhbf9S.

**Comment by AbhimanyuAryan** [comment]
have you tried installing the version of Nettle("0.5") that Genie uses?

**Comment by gameCoder101** [comment]
I think, look my previous message - [49dea1ee] + Nettle v0.5.1

**Comment by gameCoder101** [comment]
I waited for the new updated Julia, i.e. 1.7.3 for Fedora 36.
Now it is installed. Then installed Genie still the same problem.
I think I have to search for another framework :-(

julia> using Genie
[ Info: Precompiling Genie [c43c736e-a2d1-11e8-161f-af95117fbd1e]
ERROR: LoadError: InitError: could not load library "libgmpxx.so.4"
libgmpxx.so.4: cannot open shared object file: No such file or directory
Stacktrace:
 [1] __init__()
   @ GMP_jll /usr/share/julia/stdlib/v1.7/GMP_jll/src/GMP_jll.jl:36
 [2] top-level scope
   @ ~/.julia/packages/JLLWrappers/QpMQW/src/toplevel_generators.jl:188
 [3] top-level scope (repeats 2 times)
   @ none:1
during initialization of module GMP_jll
in expression starting at /home/goncalvfg/.julia/packages/Nettle_jll/LjWzs/src/wrappers/x86_64-linux-gnu.jl:4
in expression starting at /home/goncalvfg/.julia/packages/Nettle_jll/LjWzs/src/Nettle_jll.jl:2
ERROR: LoadError: Failed to precompile Nettle_jll [4c82536e-c426-54e4-b420-14f461c4ed8b] to /home/goncalvfg/.julia/compiled/v1.7/Nettle_jll/jl_Bd2IaD.
Stacktrace:
 [1] top-level scope (repeats 2 times)
   @ none:1
in expression starting at /home/goncalvfg/.julia/packages/Nettle/NhQ2s/src/Nettle.jl:1
ERROR: LoadError: Failed to precompile Nettle [49dea1ee-f6fa-5aa6-9a11-8816cee7d4b9] to /home/goncalvfg/.julia/compiled/v1.7/Nettle/jl_sSoJcy.
Stacktrace:
 [1] include(x::String)
   @ Genie ~/.julia/packages/Genie/hbOGK/src/Genie.jl:4
 [2] top-level scope
   @ ~/.julia/packages/Genie/hbOGK/src/Genie.jl:34
 [3] top-level scope (repeats 2 times)
   @ none:1
in expression starting at /home/goncalvfg/.julia/packages/Genie/hbOGK/src/Encryption.jl:1
in expression starting at /home/goncalvfg/.julia/packages/Genie/hbOGK/src/Genie.jl:1
ERROR: Failed to precompile Genie [c43c736e-a2d1-11e8-161f-af95117fbd1e] to /home/goncalvfg/.julia/compiled/v1.7/Genie/jl_hYmv4e.

**Comment by essenciary** [comment]
Do you have the manually installed Nettle in the same Project where you're `using Genie`?

**Comment by essenciary** [comment]
Oh, there you go: 
https://discourse.julialang.org/t/genie-nettle-and-nettle-jll-keep-failing-to-precompile/69081

**Comment by gameCoder101** [comment]
Great, now it works!

At least you could do a note for those who struggle to make it work in your README.

Thanks.

**Comment by AbhimanyuAryan** [comment]
This seems to have fixed with 1.7.3. Closing please re-open if the issue persists 

```[17:03:01] ~ on❯ julia +1.7.3~x64
               _
   _       _ _(_)_     |  Documentation: https://docs.julialang.org
  (_)     | (_) (_)    |
   _ _   _| |_  __ _   |  Type "?" for help, "]?" for Pkg help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 1.7.3 (2022-05-06)
 _/ |\__'_|_|_|\__'_|  |  Official https://julialang.org/ release
|__/                   |

(@v1.7) pkg> add Genie
    Updating registry at `C:\Users\aryan\.julia\registries\General.toml`
   Resolving package versions...
   Installed JuliaInterpreter ─── v0.9.17
   Installed TranscodingStreams ─ v0.9.10
   Installed JuliaFormatter ───── v1.0.16
    Updating `C:\Users\aryan\.julia\environments\v1.7\Project.toml`
  [c43c736e] + Genie v5.11.3
    Updating `C:\Users\aryan\.julia\environments\v1.7\Manifest.toml`
  [c7e460c6] + ArgParse v1.1.4
  [d1d4a3ce] + BitFlags v0.1.7
  [00ebfdb7] + CSTParser v3.3.6
  [da1fd8a2] + CodeTracking v1.1.1
  [944b1d66] + CodecZlib v0.7.0
  [a80b9123] + CommonMark v0.8.7
  [34da2185] + Compat v4.5.0
  [a8cc5b0e] + Crayons v4.1.1
  [864edb3b] + DataStructures v0.18.13
  [8f5d6c58] + EzXML v1.1.0
  [48062228] + FilePathsBase v0.9.20
  [c43c736e] + Genie v5.11.3
  [c27321d9] + Glob v1.3.0
  [cd3eb016] + HTTP v1.5.5
  [77172c1b] + HttpCommon v0.5.0
  [6d011eab] + Inflector v1.0.1
  [83e8ac13] + IniFile v0.5.1
  [692b3bcd] + JLLWrappers v1.4.1
  [682c06a0] + JSON v0.21.3
  [0f8b85d8] + JSON3 v1.12.0
  [98e50ef6] + JuliaFormatter v1.0.16
  [aa1ae85d] + JuliaInterpreter v0.9.17
  [e6f89c97] + LoggingExtras v1.0.0
  [6f1432cf] + LoweredCodeUtils v2.2.2
  [739be429] + MbedTLS v1.1.7
  [39ec1447] + Millboard v0.2.5
  [49dea1ee] + Nettle v1.0.0
  [4d1e1d77] + Nullables v1.0.0
  [4d8831e6] + OpenSSL v1.3.2
  [bac558e1] + OrderedCollections v1.4.1
  [69de0a69] + Parsers v2.5.1
  [21216c6a] + Preferences v1.3.0
  [189a3867] + Reexport v1.2.2
  [ae029012] + Requires v1.3.0
  [295af30f] + Revise v3.4.0
  [777ac1f9] + SimpleBufferStream v1.1.0
  [66db9d55] + SnoopPrecompile v1.0.1
  [69024149] + StringEncodings v0.3.5
  [856f2bd8] + StructTypes v1.10.0
  [b718987f] + TextWrap v1.0.1
  [0796e94c] + Tokenize v0.5.24
  [3bb67fe8] + TranscodingStreams v0.9.10
  [30578b45] + URIParser v0.4.1
  [5c2747f8] + URIs v1.4.1
  [ddb6d928] + YAML v0.4.8
  [94ce4f54] + Libiconv_jll v1.16.1+1
  [4c82536e] + Nettle_jll v3.7.2+0
  [458c3c95] + OpenSSL_jll v1.1.19+0
  [02c8fc9c] + XML2_jll v2.9.14+0
  [0dad84c5] + ArgTools
  [56f22d72] + Artifacts
  [2a0f44e3] + Base64
  [ade2ca70] + Dates
  [8ba89e20] + Distributed
  [f43a241f] + Downloads
  [7b1f6079] + FileWatching
  [b77e0a4c] + InteractiveUtils
  [b27032c2] + LibCURL
  [76f85450] + LibGit2
  [8f399da3] + Libdl
  [37e2e46d] + LinearAlgebra
  [56ddb016] + Logging
  [d6f4376e] + Markdown
  [a63ad114] + Mmap
  [ca575930] + NetworkOptions
  [44cfe95a] + Pkg
  [de0858da] + Printf
  [3fa0cd96] + REPL
  [9a3f8284] + Random
  [ea8e919c] + SHA
  [9e88b42a] + Serialization
  [6462fe0b] + Sockets
  [fa267f1f] + TOML
  [a4e569a6] + Tar
  [8dfed614] + Test
  [cf7118a7] + UUIDs
  [4ec0a83e] + Unicode
  [e66e0078] + CompilerSupportLibraries_jll
  [781609d7] + GMP_jll
  [deac9b47] + LibCURL_jll
  [29816b5a] + LibSSH2_jll
  [c8ffd9c3] + MbedTLS_jll
  [14a3606d] + MozillaCACerts_jll
  [4536629a] + OpenBLAS_jll
  [83775a58] + Zlib_jll
  [8e850b90] + libblastrampoline_jll
  [8e850ede] + nghttp2_jll
  [3f19e933] + p7zip_jll
Precompiling project...
  56 dependencies successfully precompiled in 26 seconds

julia> import Genie
```
